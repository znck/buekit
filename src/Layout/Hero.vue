<script>
import { cssModule, mergeData } from '../utils'
import { colors, sizes, createTag } from '../mixins'

export default {
  name: 'Hero',
  functional: true,
  mixins: [colors, sizes, createTag('section')],
  props: {
    bold: {
      default: false,
      type: Boolean
    }
  },
  render (h, ctx) {
    const s = cssModule(ctx.$style)
    const classes = [s('hero')]
    const { bold, color, size, tag } = ctx.props
    const slots = ctx.slots()

    if (color) classes.push(s('is-' + color))
    if (size) classes.push(s('is-' + size))
    if (bold) classes.push(s('is-bold'))

    return h(tag, mergeData({ class: classes }, ctx.data), [
      slots.head && slots.head.length && h('div', { class: s('hero-head') }, slots.head),
      h('div', { class: s('hero-body') }, slots.default),
      slots.foot && slots.foot.length && h('div', { class: s('hero-foot') }, slots.foot)
    ])
  }
}
</script>

<style lang="scss" module>
@import '~bulma/sass/utilities/_all.sass';
@import '../_base.scss';
@import '~bulma/sass/layout/hero.sass';
</style>

