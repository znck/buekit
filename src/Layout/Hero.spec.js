import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'

import Hero from './Hero.vue'

function snapshots (options) {
  options.forEach(([props = {}, slots = {}]) =>
    it ('matches snapshot', () => {
      const wrapper = shallow(Hero, { props, slots})
      const renderer = createRenderer()
  
      renderer.renderToString(wrapper.vm, (err, str) => {
        if (err) throw new Error(err)
        expect(str).toMatchSnapshot()
      })
    })
  )
}

describe('Hero.vue', () => {
  snapshots ([
    [{ size: 'medium' }, { default: 'Content' }],
    [{ size: 'medium' }, { default: 'Content', head: 'Head' }],
    [{ size: 'medium' }, { default: 'Content', head: 'Head', foot: 'Hoot' }],
    [{ size: 'medium', bold: true }, { default: 'Content' }],
    [{ tag: 'article' }, { default: 'Content' }],
    [{ color: 'primary' }, { default: 'Content' }],
  ])
})
