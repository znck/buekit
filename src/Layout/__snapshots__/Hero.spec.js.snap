// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero.vue matches snapshot 1`] = `
<section class="hero">
    <div class="hero-body"></div>
</section>
`;

exports[`Hero.vue matches snapshot 2`] = `
<section class="hero">
    <div class="hero-body"></div>
</section>
`;

exports[`Hero.vue matches snapshot 3`] = `
<section class="hero">
    <div class="hero-body"></div>
</section>
`;

exports[`Hero.vue matches snapshot 4`] = `
<section class="hero">
    <div class="hero-body"></div>
</section>
`;

exports[`Hero.vue matches snapshot 5`] = `
<section class="hero">
    <div class="hero-body"></div>
</section>
`;

exports[`Hero.vue matches snapshot 6`] = `
<section class="hero">
    <div class="hero-body"></div>
</section>
`;
